### PR TITLE
[WAIT] Remove executor and use official ES image

### PR DESCRIPTION
**Pull request by philwinder**

### CODE REVIEW

**Comment by philwinder** [comment]
Jenkins passes: 
- https://ci.infra.container-solutions.com/view/All/job/ElasticsearchPullRequests/269/console
- https://ci.infra.container-solutions.com/view/All/job/ElasticsearchPullRequests/270/console

File: docs/index.md [hunk 1]
```diff
@@ -156,19 +151,13 @@ Usage: (Options preceded by an asterisk are required) [options]
...
-       Default: 30000
     --executorImage
-       The docker executor image to use. [DOCKER MODE ONLY]
-       Default: mesos/elasticsearch-executor
+       The docker executor image to use. E.g. 'elasticsearch:latest' [DOCKER
+       MODE ONLY]
+       Default: elasticsearch:latest
     --executorName
        The name given to the executor task.
        Default: elasticsearch-executor
```
Comment (frankscholten) [inline docs/index.md]: Rename to --esTaskName?

File: scheduler/src/main/java/org/apache/mesos/elasticsearch/scheduler/OfferStrategy.java [hunk 2]
```diff
@@ -39,11 +43,18 @@ private boolean isHostnameResolveable(String hostname) {
...
         } else {
-            return clusterState.getStatus(taskList.get(0).getTaskId()).getStatus().getState().equals(Protos.TaskState.TASK_RUNNING);
+            try {
+                // TODO (PNW): Better get HTTP address method. We do this everywhere.
+                InetSocketAddress clientAddress = taskList.get(clusterState.getTaskList().get(0).getTaskId().getValue()).getClientAddress();
+                String hostAddress = NetworkUtils.addressToString(clientAddress, configuration.getIsUseIpAddress());
+                return Unirest.get(hostAddress).asString().getStatus() == 200;
+            } catch (UnirestException e) {
+                return false;
+            }
```
Comment (frankscholten) [inline scheduler/src/main/java/org/apache/mesos/elasticsearch/scheduler/OfferStrategy.java]: Let's add a refactoring task to fix this code duplication

File: scheduler/src/main/java/org/apache/mesos/elasticsearch/scheduler/TaskInfoFactory.java [hunk 3]
```diff
@@ -33,8 +35,12 @@
     private static final Logger LOGGER = Logger.getLogger(TaskInfoFactory.class);
 
     public static final String TASK_DATE_FORMAT = "yyyyMMdd'T'HHmmss.SSS'Z'";
+    public static final String CONTAINER_PATH_DATA = "/usr/share/elasticsearch/data";
+    public static final String CONTAINER_PATH_CONF = "/usr/share/elasticsearch/config";
+    public static final String HOST_SANDBOX = "./."; // Do to some protobuf weirdness. Requires './.' Not just '.'
+    public static final String HOST_PATH_HOME = HOST_SANDBOX + "/es_home";
```
Comment (frankscholten) [inline scheduler/src/main/java/org/apache/mesos/elasticsearch/scheduler/TaskInfoFactory.java]: Due

File: scheduler/src/main/java/org/apache/mesos/elasticsearch/scheduler/TaskInfoFactory.java [hunk 4]
```diff
@@ -50,88 +56,182 @@ public TaskInfoFactory(ClusterState clusterState) {
...
+        String mkdir = "mkdir -p " + folders + "; ";
+        String chown = "chown -R nobody:nogroup " + folders + "; ";
+        String command = mkdir +
+                        chown +
+                " su -s /bin/sh -c \""
+                        + Configuration.ES_BINARY
+                        + " "
+                        + args.stream().collect(Collectors.joining(" "))
+                        + "\" nobody";
+        final Protos.Environment environment = Protos.Environment.newBuilder().addAllVariables(new ExecutorEnvironmentalVariables(configuration).getList()).build();
```
Comment (frankscholten) [inline scheduler/src/main/java/org/apache/mesos/elasticsearch/scheduler/TaskInfoFactory.java]: Can the command generation code be moved to a method in the Configuration class? The command uses address, Configuration.BINARY, Configuration.ES_TAR and datadir so I think it should be moved there.

File: scheduler/src/main/java/org/apache/mesos/elasticsearch/scheduler/TaskInfoFactory.java [hunk 5]
```diff
@@ -140,30 +240,51 @@ public ByteString toData(String hostname, String ipAddress, ZonedDateTime zonedD
...
+        args.add("--default.network.publish_host=_non_loopback:ipv4_");
+        args.add("--default.gateway.recover_after_nodes=1");
+        args.add("--default.gateway.expected_nodes=1");
+        args.add("--default.indices.recovery.max_bytes_per_sec=100mb");
+        args.add("--default.discovery.type=zen");
+        args.add("--default.discovery.zen.fd.ping_timeout=30s");
+        args.add("--default.discovery.zen.fd.ping_interval=1s");
+        args.add("--default.discovery.zen.fd.ping_retries=30");
+        args.add("--default.discovery.zen.ping.multicast.enabled=false");
+
```
Comment (frankscholten) [inline scheduler/src/main/java/org/apache/mesos/elasticsearch/scheduler/TaskInfoFactory.java]: I would move this to Configuration as well

File: scheduler/src/test/java/org/apache/mesos/elasticsearch/scheduler/TaskInfoFactoryTest.java [hunk 6]
```diff
@@ -134,6 +129,7 @@ public void testCreateTaskInfo() {
                                             .build();
     }
 
+    // TODO (PNW): Reinstate this test when Jar mode is added.
     @Test
```
Comment (frankscholten) [inline scheduler/src/test/java/org/apache/mesos/elasticsearch/scheduler/TaskInfoFactoryTest.java]: What does this TODO mean? The test is not ignored.

File: docs/index.md (hunk 1 above)
Comment (philwinder) [inline docs/index.md]: Not for now. Maintain backwards compatibility and ensure people know they are executors.

File: scheduler/src/main/java/org/apache/mesos/elasticsearch/scheduler/OfferStrategy.java (hunk 2 above)
Comment (philwinder) [inline scheduler/src/main/java/org/apache/mesos/elasticsearch/scheduler/OfferStrategy.java]: Done. #504

**Comment by philwinder** [comment]
Jenkins pass: https://ci.infra.container-solutions.com/view/All/job/ElasticsearchPullRequests/280/console

**Pull request merged**